File: src/SIL.Machine.TS/src/web-api/translation-result-dto.ts
import { TranslationSources } from '../translation/translation-sources';
import { AlignedWordPairDto } from './aligned-word-pair-dto';
import { PhraseDto } from './phrase-dto';

export interface TranslationResultDto {
  target: string[];
  confidences: number[];
  sources: TranslationSources[];
  alignment: AlignedWordPairDto[];
  phrases: PhraseDto[];
}

File: src/Serval.Core/TranslationResultDto.cs
namespace Serval.Core
{
    public class TranslationResultDto
    {
        public string Translation { get; set; }

        public string[] Tokens { get; set; }

        public float[] Confidences { get; set; }

        public TranslationSources[] Sources { get; set; }

        public AlignedWordPairDto[] Alignment { get; set; }

        public PhraseDto[] Phrases { get; set; }
    }
}
